x-pack/plugins/maps/common/descriptor_types/map_descriptor.ts

```diff
@@ -68,8 +68,13 @@ export type TooltipState = {
 
 export type DrawState = {
   actionId: string;
-  drawType: DRAW_TYPE;
+  drawShape?: DRAW_SHAPE;
   filterLabel?: string; // point radius filter alias
   geometryLabel?: string;
   relation?: ES_SPATIAL_RELATIONS;
 };
+
+export type EditState = {
+  layerId?: string;
```

Comment:
Unlike with filters, each of these is updated at different points. The `layerId` starts off `undefined` and is updated when a user clicks on a layer and goes to `Edit features`, the `drawShape` is `undefined` then updated when a user selects the shape to draw from the toolbar that pops up on the left.

Reply:
Since EditState is useless without a layerId, how about just setting "editState: { layerId }" when a user clicks on the edit layer action. And then set "editState: undefined" when a user leaves editing mode. editState is already defined as "editState?: EditState" so you have to handle undefined for editState.

Reply:
Yep, this is cleaner. Also exposed a bug while I was working on it where switching between editing point and shape layers allowed attempting to draw shapes on point layers. This has been fixed.